desc/geometry/surface.py

```diff
@@ -519,6 +519,83 @@ def from_values(
         )
         return surf
 
+    @classmethod
+    def from_shape_parameters(
+        cls,
+        major_radius=10.0,
+        minor_radius=1.0,
+        elongation=1.0,
+        triangularity=0.3,
+        squareness=0.0,
+        eccentricity=0.0,
+        torsion=0.0,
+        twist=0.0,
+        NFP=1,
+        sym=True,
+    ):
+        """Create a surface using a generalized Miller parameterization.
+
+        Parameters
+        ----------
+        major_radius : float > 0
+            Average major radius. Constant term in the R coordinate.
+        minor_radius : float > 0
```

Comment:
is it better to use aspect ratio here?

Reply:
instead of minor radius? yea maybe, they are basically the same but aspect ratio is something that would actually be scanned over and something people think of more than minor radius